app/models/post.rb

```diff
@@ -57,6 +57,34 @@ def address
     [street, city, state, postal_code, country].compact.join(", ")
   end
 
+  def self.filter_type(type)
+    if type == "needed" || type == "available"
+      send(type.to_sym)
+    else
+      all
+    end
+  end
+
+  def self.filter_near(location)
+    results = near(location)
+
+    if results.blank?
```

Comment:
Isn't this going to return everything if there is nothing within the range? Wouldn't we prefer a 'there were no results' in this case and return something else here?